HyperPlay-Gaming/hyperplay-desktop-client
# [Feat] Add Hardware Wallet Support

- Adds hardware wallet support for MetaMask.
- Upgraded electron to 23.1.3 which now matches heroic
- Refactors window management
- Removes mm animation due to the combination of mobx and useEffect calling `<webview>.reload()` to throw 
`Error: Invalid guestInstanceId`

### To Test
1. Connect your hardware wallet through usb if ledger or trezor
2. Connect with MM extension in HyperPlay
3. Click the ... and connect hardware wallet
<img width="305" alt="image" src="https://user-images.githubusercontent.com/27568879/232196196-6edcfd34-1b1f-43cf-b1b0-3cfcb7f47a5a.png">
4. Follow the on screen prompts. See screenshots below for expected behavior


## Screenshots
<img width="1263" alt="image" src="https://user-images.githubusercontent.com/27568879/232195806-8307def7-e61c-4d92-9da2-39612d91c0a7.png">

### Trezor

<img width="1255" alt="image" src="https://user-images.githubusercontent.com/27568879/232195847-a3015528-da2b-4b9c-8d6f-18b18f94c962.png">
<img width="1259" alt="image" src="https://user-images.githubusercontent.com/27568879/232195878-f1578153-8ef5-434a-8b60-97fbe619b64a.png">
<img width="1259" alt="image" src="https://user-images.githubusercontent.com/27568879/232195913-57e2f3ea-3153-417a-8c55-36a23f5c4684.png">

### Ledger

<img width="1261" alt="image" src="https://user-images.githubusercontent.com/27568879/232196131-890e77e1-f21a-4c10-a8ba-b7f4bbc2cd99.png">

### Lattice

<img width="1259" alt="image" src="https://user-images.githubusercontent.com/27568879/232196161-9119a4fb-b16d-4340-b165-da54e5f557fb.png">


---

Use the following Checklist if you have changed something on the Backend or Frontend:
